commons/utils.js

```diff
@@ -66,6 +67,116 @@ const getUnit = bridgeMode => {
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorAdded'
+    event.returnValues.validator = address
+  } else if (
+    event.event === undefined &&
+    event.raw &&
+    event.raw.topics &&
+    event.raw.topics[0] === '0xe1434e25d6611e0db941968fdc97811c982ac1602e951637d206f5fdda9dd8f1'
+  ) {
+    const rawAddress = event.raw.data === '0x' ? event.raw.topics[1] : event.raw.data
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorRemoved'
+    event.returnValues.validator = address
+  }
+}
+
+const processValidatorsEvents = events => {
+  const validatorList = new Set()
+  events.forEach(event => {
+    parseValidatorEvent(event)
+
+    if (event.event === 'ValidatorAdded') {
+      validatorList.add(event.returnValues.validator)
+    } else if (event.event === 'ValidatorRemoved') {
+      validatorList.delete(event.returnValues.validator)
+    }
+  })
+
+  return Array.from(validatorList)
+}
+
+const tryCall = async (method, fallbackValue) => {
+  try {
+    return await method.call()
+  } catch (e) {
+    return fallbackValue
+  }
+}
+
+const getDeployedAtBlock = async contract => tryCall(contract.methods.deployedAtBlock(), 0)
+
+const getPastEvents = async (contract, { event, fromBlock, toBlock, options }) => {
+  let events
+  try {
+    events = await contract.getPastEvents(event || 'allEvents', {
+      options: options || {},
+      fromBlock,
+      toBlock: toBlock || 'latest'
+    })
+  } catch (e) {
+    if (e.message.includes('query returned more than 1000 results') && Number.isInteger(toBlock)) {
+      const middle = fromBlock.add(toBlock).divRound(toBN(2))
+      const middlePlusOne = middle.add(toBN(1))
+
+      const firstHalfEvents = await getPastEvents(contract, {
+        event: event || 'allEvents',
+        fromBlock,
+        toBlock: middle,
+        options
+      })
+      const secondHalfEvents = await getPastEvents(contract, {
+        event: event || 'allEvents',
+        fromBlock: middlePlusOne,
+        toBlock,
+        options
+      })
+      events = [...firstHalfEvents, ...secondHalfEvents]
+    } else {
+      throw new Error(e)
+    }
+  }
+  return events
+}
+
+const getValidatorList = async (address, eth, options) => {
+  options.logger && options.logger.debug && options.logger.debug('getting validatorList')
+
+  const validatorsContract = new eth.Contract(REWARDABLE_VALIDATORS_ABI, address) // in monitor, BRIDGE_VALIDATORS_ABI was used
+  const validators = await tryCall(validatorsContract.methods.validatorList(), [])
+
+  if (validators.length) {
+    return validators
+  }
+
+  options.logger && options.logger.debug && options.logger.debug('getting validatorsEvents')
+
+  const deployedAtBlock = await tryCall(validatorsContract.methods.deployedAtBlock(), 0)
+  const fromBlock = options.fromBlock || Number(deployedAtBlock) || 0
+  const toBlock = options.toBlock || 'latest'
+
+  const validatorsEvents = await getPastEvents({
```

Comment:
This `getPastEvents` call should be updated to match the signature of the method

```javascript
const getPastEvents = async (contract, { event, fromBlock, toBlock, options }) => {
```
For example:
```javascript
  const validatorsEvents = await getPastEvents(new eth.Contract([], address), {
    event: 'allEvents',
    fromBlock,
    toBlock,
    options: {}
  })
```

Reply:
Right. Updated.